Lib/argparse.py

```diff
@@ -901,6 +904,24 @@ def __init__(self,
                 option_string = '--no-' + option_string[2:]
                 _option_strings.append(option_string)
 
+        # We need `_deprecated` special value to ban explicit arguments that
+        # match default value. Like:
+        #   parser.add_argument('-f', action=BooleanOptionalAction, type=int)
+        for field_name in ('type', 'choices', 'metavar'):
+            if locals()[field_name] is not _deprecated_default:
+                warnings._deprecated(
+                    field_name,
+                    "{name!r} is deprecated as of Python 3.12 and will be "
+                    "removed in Python {remove}.",
+                    remove=(3, 14))
+
+        if type is _deprecated_default:
+            type = None
+        if choices is _deprecated_default:
+            choices = None
+        if metavar is _deprecated_default:
+            metavar = None
```

Comment:
Could we remove these three if checks and reset all three back to `None`, with a comment?

Reply:
It feels safer not to do that. Right now my change does not change any behavior (for subclasses, etc). But, setting these 3 params to `None` without conditions might change it for some corner cases. So, I would rather not :)